### ISSUE TITLE
[Core] Ray fails with `AttributeError: module 'pydantic._internal' has no attribute '_model_construction'` when pydantic installed

### ISSUE THREAD

**Issue by dmatlak** [ray-project/ray#41913]
### What happened + What you expected to happen

As in topic. Error is raised when calling `ray.init()`:
```
2023-12-14 09:29:28,002 ERROR services.py:1329 -- Failed to start the dashboard , return code 1
2023-12-14 09:29:28,002 ERROR services.py:1354 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure' to find where the log file is.
2023-12-14 09:29:28,003 ERROR services.py:1398 -- 
The last 20 lines of /tmp/ray/session_2023-12-14_09-29-26_578182_2395000/logs/dashboard.log (it contains the error message from the dashboard): 
  File "/home/xxx/.pyenv/versions/3.8.16/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/xxx/ray_issue/lib/python3.8/site-packages/ray/dashboard/modules/job/cli.py", line 16, in <module>
    from ray.job_submission import JobStatus, JobSubmissionClient
  File "/home/xxx/ray_issue/lib/python3.8/site-packages/ray/job_submission/__init__.py", line 2, in <module>
    from ray.dashboard.modules.job.pydantic_models import DriverInfo, JobDetails, JobType
  File "/home/xxx/ray_issue/lib/python3.8/site-packages/ray/dashboard/modules/job/pydantic_models.py", line 4, in <module>
    from ray._private.pydantic_compat import BaseModel, Field, PYDANTIC_INSTALLED
  File "/home/xxx/ray_issue/lib/python3.8/site-packages/ray/_private/pydantic_compat.py", line 100, in <module>
    monkeypatch_pydantic_2_for_cloudpickle()
  File "/home/xxx/ray_issue/lib/python3.8/site-packages/ray/_private/pydantic_compat.py", line 58, in monkeypatch_pydantic_2_for_cloudpickle
    pydantic._internal._model_construction.SchemaSerializer = (
AttributeError: module 'pydantic._internal' has no attribute '_model_construction'
```

### Versions / Dependencies

OS: Ubuntu 22.04
Python: Python 3.8
ray==2.8.1
pydantic==2.5.2

### Reproduction script

```shell
python -m venv ray_issue
source ray_issue/bin/activate
pip install pydantic=="2.5.2" ray=="2.8.1"
python -c "import ray; ray.init()"
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by silverbulletmdc** [comment]
+1

**Comment by shrekris-anyscale** [comment]
Pydantic 2 is not supported in Ray 2.8. Please downgrade to `pydantic<2`.

Starting in Ray 2.9 (which should be released around the end of the month), `pydantic>=2.5.0` will be compatible with Ray.

If you want to start using Pydantic 2 ASAP, then the [Ray nightly wheel](https://docs.ray.io/en/releases-2.8.1/ray-overview/installation.html#daily-releases-nightlies) is already compatible with it. I would recommend installing the nightly and using `pydantic>=2.5.0`.

**Comment by silverbulletmdc** [comment]
But gradio 4 relies on pydantic2, so ray2.8 is not compitable with gradio currently. Hope the 2.9 be released as soon as possible.

**Comment by shrekris-anyscale** [comment]
@silverbulletmdc thanks for clarifying. Could you try your Gradio application with the Ray nightly to make sure that it works?

**Comment by silverbulletmdc** [comment]
It works well.

**Comment by edoakes** [comment]
Closing as this is confirmed fixed in Ray 2.9+ (releasing coming in the coming weeks)
